### PR TITLE
Support traffic validation report generation

### PR DESCRIPTION
**Pull request by raych1**
Report including coverage data, error information.

### CODE REVIEW

File: lib/report/generateReport.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    this.language = language;
+    this.generalErrorResults = new Map();
+    this.setMetrics();
+    this.sortOperationIds();
+    this.prepareDataForRendering();
+  }
+
+  private prepareDataForRendering() {
+    const specLinkPrefix = "https://github.com/Azure/azure-rest-api-specs/blob/main/";
+    const payloadLinkPrefix = "https://github.com/scbedd/oav-traffic-converter/blob/main/sample-tables-input/"
```
Comment (ruowan) [inline lib/report/generateReport.ts]: Why hard code this here? can this be a argument?

Comment (ruowan) [inline lib/report/generateReport.ts]: Why hard code this here? can this be a argument?

File: lib/report/generateReport.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    this.generatedDate = new Date();
+    this.language = language;
+    this.generalErrorResults = new Map();
+    this.setMetrics();
+    this.sortOperationIds();
+    this.prepareDataForRendering();
+  }
+
+  private prepareDataForRendering() {
+    const specLinkPrefix = "https://github.com/Azure/azure-rest-api-specs/blob/main/";
```
Comment (xiaoxuqi-ms) [inline lib/report/generateReport.ts]: Is it better to get these prefix from arguments?

File: lib/swaggerValidator/trafficValidator.ts [hunk 3]
```diff
@@ -216,15 +230,27 @@ export class TrafficValidator {
...
     this.operationSpecMapper.forEach((value: string[], key: string) => {
+      unCoveredOperationsList = [];
       if (this.trafficOperation.get(key) === undefined) {
         coveredOperaions = 0;
         coverageRate = 0;
         this.coverageData.set(key, 0);
       } else if (value !== undefined && value.length !== 0) {
-        coveredOperaions = this.trafficOperation.get(key)!.length;
+        const validatedOperations = this.trafficOperation.get(key);
+        coveredOperaions = validatedOperations!.length;
```
Comment (xiaoxuqi-ms) [inline lib/swaggerValidator/trafficValidator.ts:243]: use 'this.validatedOperations!.length;' ?

File: lib/swaggerValidator/trafficValidator.ts [hunk 4]
```diff
@@ -216,15 +230,27 @@ export class TrafficValidator {
     let coveredOperaions: number;
     let coverageRate: number;
     let validationFailOperations: number;
+    let unCoveredOperationsList: OperationMeta[];
     this.operationSpecMapper.forEach((value: string[], key: string) => {
+      unCoveredOperationsList = [];
       if (this.trafficOperation.get(key) === undefined) {
```
Comment (xiaoxuqi-ms) [inline lib/swaggerValidator/trafficValidator.ts:237]: if there is no operations found from traffic, should we also list the uncovered operations?

File: lib/report/generateReport.ts (hunk 1 above)
Comment (raych1) [reply lib/report/generateReport.ts]: Exposed this command line parameter.

Comment (raych1) [reply lib/report/generateReport.ts]: Exposed this command line parameter.

**Review by ruowan — APPROVED** [review]

**Review by xiaoxuqi-ms — APPROVED** [review]

File: lib/swaggerValidator/trafficValidator.ts (hunk 4 above)
Comment (xiaoxuqi-ms) [reply lib/swaggerValidator/trafficValidator.ts:237]: For single RP validation, if there are no operations found from traffic for a certain swagger file, then the missing operations should be reported. But if it is multi-RP scenario, we can ignore this.

File: lib/swaggerValidator/trafficValidator.ts (hunk 3 above)
Comment (raych1) [reply lib/swaggerValidator/trafficValidator.ts:243]: there's no this.validatedOperations defined.

File: lib/swaggerValidator/trafficValidator.ts (hunk 4 above)
Comment (raych1) [reply lib/swaggerValidator/trafficValidator.ts:237]: When there's no operations found from traffic, uncovered operations is also reported. It works for both single RP and multiple RP.

**Pull request merged**